test/arrayops.jl

```diff
@@ -2908,6 +2908,11 @@ end
     @inferred accumulate(+, randn(3))
     @inferred accumulate(+, randn(3); init=1)
 
+    # inbounds propagation
+    Base.@propagate_inbounds op_bounds(a, b) = (1, 2)[a] + (1, 2)[b]
+    @test_throws BoundsError accumulate(op_bounds, 1:10)
+    @test_throws BoundsError Base.accumulate_pairwise(op_bounds, 1:10)
```

Comment:
This will always pass because tests are run with bounds checked :)

You can put this in boundscheck_exec to explicitly test it when checkbounds isn't entirely disabled.

You could also just test with `getindex` directly:

```julia
julia> accumulate(getindex, [1:10,2:10,2:10,2:10])
4-element Vector{UnitRange{Int64}}:
 1:10
 2:10
 3:11
 4:12

julia> accumulate((x,i)->getindex(x,i), [1:10,2:10,2:10,2:10])
ERROR: BoundsError: attempt to access 9-element UnitRange{Int64} at index [2:10]
```

Reply:
ah good point. in fact it looks like this PR doesn't yet fix `accumulate(getindex, [1:10,2:10,2:10,2:10])` to begin with